staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,8 +180,17 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// The CachePreprocessHandler is called for each object which is about to
+	// be stored into the cache.
+	// This function is intended for you to take the opportunity to
+	// remove, transform, or normalize fields. One use case is to strip unused
+	// metadata fields out of objects to save on RAM cost.
+	SetPreprocessHandler(handler CachePreprocessHandler) error
```

Comment:
Can you unify your change with the TransformingInformer we already have:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/controller.go#L395

It seems you're trying to achieve pretty much the same thing.

Reply:
@wojtek-t thanks for looking. I've modified the PR to refactor SharedIndexInformer to make use of the `newInformer` constructor rather than constructing its `controller` itself. Please take another look!